consent-ws/src/main/java/org/broadinstitute/consent/http/service/DatabaseMatchingServiceAPI.java

```diff
@@ -105,24 +101,29 @@ public Match findMatchForPurpose(String purposeId){
     }
 
     private Match singleEntitiesMatch(Consent consent, Document dar) throws Exception {
+        if (consent == null) {
```

Comment:
I like the pattern of checking preconditions at the top of a method, I think this is a nice improvement